Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -103,27 +117,49 @@ + (void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
 
 + (void)didReceiveRemoteNotification:(NSDictionary *)notification
 {
+  NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:notification];
+  UIApplication* application = [UIApplication sharedApplication];
+  if (application.applicationState == UIApplicationStateActive) {
+    userInfo[@"applicationState"] = @"active";
+  } else {
+    userInfo[@"applicationState"] = @"background";
+  }
+
   [[NSNotificationCenter defaultCenter] postNotificationName:RCTRemoteNotificationReceived
                                                       object:self
-                                                    userInfo:notification];
+                                                    userInfo:userInfo];
 }
 
 + (void)didReceiveLocalNotification:(UILocalNotification *)notification
 {
-  NSMutableDictionary *details = [NSMutableDictionary new];
+  NSDictionary *baseNotificationData = [RCTPushNotificationManager extractLocalNotificationData:notification 
+                                                                           withActionIdentifier:nil];
+  NSMutableDictionary *details = [NSMutableDictionary dictionaryWithDictionary:baseNotificationData];
   if (notification.alertBody) {
     details[@"alertBody"] = notification.alertBody;
   }
   if (notification.userInfo) {
     details[@"userInfo"] = RCTJSONClean(notification.userInfo);
   }
+
+  UIApplication* application = [UIApplication sharedApplication];
```

Comment:
Ah I see: https://developer.apple.com/library/ios/documentation/General/Conceptual/ExtensibilityPG/ 
But yes, I think it still shouldn't matter in this file.